test/Microsoft.AspNet.Mvc.Xml.Test/XmlAssert.cs

```diff
@@ -42,8 +70,24 @@ private static XElement SortAttributes(XElement element)
         {
             return new XElement(
                     element.Name,
+                    element.Nodes().OfType<XText>().FirstOrDefault(),
```

Comment:
these uses of `FirstOrDefault()` will cause the equality checks to ignore additional text nodes.  need to include _all_ text nodes.  probably fine that you're ignoring comments and processing instructions but text nodes are (all) important.